handler/src/main/java/io/netty/handler/pcap/PCapWriter.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2020 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.pcap;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.OutputStream;
+
+final class PCapWriter implements Closeable {
+    private final long myStartTime = System.nanoTime();
+    private final OutputStream outputStream;
+
+    /**
+     * This uses {@link OutputStream} for writing Pcap.
+     * Pcap Global Header is also written on construction.
+     *
+     * @throws IOException If {@link OutputStream#write(byte[])} throws an exception
+     */
+    PCapWriter(OutputStream outputStream) throws IOException {
+        this.outputStream = outputStream;
+
+        ByteBuf byteBuf = Unpooled.buffer();
+        io.netty.handler.codec.pcap.PcapHeaders.writeGlobalHeader(byteBuf);
+        byteBuf.readBytes(outputStream, 24); // Pcap Global Header size is of 24 Bytes
+        byteBuf.release();
+    }
+
+    /**
+     * Write Packet in Pcap OutputStream.
+     *
+     * @param packetHeaderBuf Packer Header {@link ByteBuf}
+     * @param packet Packet
+     * @throws IOException If {@link OutputStream#write(byte[])} throws an exception
+     */
+    void writePacket(ByteBuf packetHeaderBuf, ByteBuf packet) throws IOException {
+        long difference = System.nanoTime() - myStartTime;
+
+        io.netty.handler.codec.pcap.PcapHeaders.writePacketHeader(
+                packetHeaderBuf,
+                (int) difference / 1000000000,
+                (int) difference / 1000000,
+                packet.readableBytes(),
+                packet.readableBytes()
+        );
+
+        packetHeaderBuf.readBytes(outputStream, 16); // Pcap Packet Header size is of 24 Bytes
```

Comment:
so either the comment is wrong or `16` is wrong ... Also wouldn't it make more sense to use `packetHeaderBuf.readableBytes()` ?

Reply:
Fixed :)